topo/node/cisco/cisco_test.go

```diff
@@ -656,3 +637,112 @@ func TestNew(t *testing.T) {
 		})
 	}
 }
+
+func TestNodeStatus(t *testing.T) {
+
+	ki := fake.NewSimpleClientset(&corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod1",
+		},
+		Status: corev1.PodStatus{
+			Phase: corev1.PodRunning,
+		},
+	})
+	ki.PrependReactor("get", "pods", func(action ktest.Action) (handled bool, ret runtime.Object, err error) {
+		if action.GetSubresource() == "log" {
+			// TODO: fake the log action
+		}
+		obj := &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "pod1",
+			},
+			Status: corev1.PodStatus{
+				Phase: corev1.PodRunning,
+			},
+		}
+		return true, obj, nil
+	})
+
+	node8000e := &node.Impl{
+		KubeClient: ki,
+		Namespace:  "test",
+		Proto: &tpb.Node{
+			Name:   "pod1",
+			Vendor: tpb.Vendor_CISCO,
+			Config: &tpb.Config{},
+			Model:  "8201-32FH",
+		},
+	}
+
+	nodeXRD := &node.Impl{
+		KubeClient: ki,
+		Namespace:  "test",
+		Proto: &tpb.Node{
+			Name:   "pod1",
+			Vendor: tpb.Vendor_CISCO,
+			Config: &tpb.Config{},
+			Model:  ModelXRD,
+		},
+	}
+
+	tests := []struct {
+		desc   string
+		status node.Status
+		ni     *node.Impl
+		log    string
+	}{
+		{
+			// successful status for 8000e Node
+			desc:   "Status test for 8000e Node",
+			status: node.StatusRunning,
+			ni:     node8000e,
+			log:    "\n ... Router up ... \n",
+		},
+		{
+			// unsuccessful status for 8000e Node
+			desc:   "Negative Status test for 8000e Node",
+			status: node.StatusPending,
+			ni:     node8000e,
+			log:    "\n ... anything other that expected message ... \n",
+		},
+		{
+			// status for xrd Node
+			desc:   "Status test for XRD Node",
+			status: node.StatusRunning,
+			ni:     nodeXRD,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			ctx := context.Background()
+			nImpl, _ := New(tt.ni)
+			n, _ := nImpl.(*Node)
+			status, err := n.Status(ctx)
```

Comment:
by moving the clientset in here you would get a more accurate test of how status works, currently status always passes and the additional behavior is tested separately. In this way the direct logic of status is not fully covered. in case the impl changes, this test would need to undergo lots of change

Reply:
agree, as i told above i have an issue to mocking log action. if u have a sample code that would be a bug help to improve this.